### PR TITLE
chore: Organize some search components

### PR DESCRIPTION
**Pull request by filipewl**
## What's the purpose of this pull request?

It's to address some of the visual inconsistencies in the search components `History`, `SuggestionsTopSearch`, and `Suggestions`.

## How does it work?

Fix clickable areas, unnecessary duplicated CSS, padding, fonts, etc.

Before|After
-|-
History-1|History-2
TopSearch-1|TopSearch-2
Suggestions-1|Suggestions-2

## How to test it?

Check the stories for these components [here](https://gatsby-store-git-refactor-search-history-n-top-faststore.vercel.app/?path=/story/organisms-search-history--default).

### CODE REVIEW

File: src/components/search/SuggestionProductCard/suggestion-product-card.scss [hunk 1]
```diff
@@ -3,6 +3,10 @@
 .suggestion-product-card {
   padding: var(--fs-spacing-1) 0;
 
+  [data-card-image] {
+    display: flex;
```
Comment (filipewl) [inline src/components/search/SuggestionProductCard/suggestion-product-card.scss:7]: Fix `[data-card-image]` having more height (62px) than needed (56px) from the image it wraps. 
Is there any other way to do it?

CleanShot 2022-04-25 at 20 58 54

File: src/components/search/History/SearchHistory.tsx [hunk 2]
```diff
@@ -20,36 +21,33 @@ const doSearch = (term: string) => {
...
+      <div data-fs-search-suggestion-header>
+        <p data-fs-search-suggestion-title>History</p>
         <Button variant="tertiary" onClick={onClear}>
           Clear
         </Button>
       </div>
       <UIList variant="ordered">
         {searchHistory.map((item, index) => (
-          <li data-store-search-history-item key={index}>
+          <li key={index} data-fs-search-suggestion-item>
```
Comment (saranicoly) [inline src/components/search/History/SearchHistory.tsx:37]: What do you think about using another key, instead of the index here? In this case, I think the key must be unique (maybe an ID or something like that). 

[This article](https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318) explains that a little bit.

File: src/components/search/Suggestions/Suggestions.tsx [hunk 3]
```diff
@@ -121,12 +121,12 @@ const Suggestions = forwardRef<HTMLDivElement, SuggestionsProps>(
...
         {SUGGESTED_PRODUCTS.length > 0 && (
-          <div className="suggestions__section">
-            <p className="suggestions__title">Suggested Products</p>
+          <div data-fs-search-suggestion-section>
+            <p data-fs-search-suggestion-title="small">Suggested Products</p>
             <UIList>
               {SUGGESTED_PRODUCTS.slice(0, MAX_SUGGESTED_PRODUCTS).map(
                 (product, index) => (
-                  <li key={index} className="suggestions__item">
+                  <li key={index} data-fs-search-suggestion-item>
```
Comment (saranicoly) [inline src/components/search/Suggestions/Suggestions.tsx:129]: Same suggestion about not using the index as a key here.

File: src/components/search/Suggestions/SuggestionsTopSearch.tsx [hunk 4]
```diff
@@ -33,14 +33,15 @@ const SuggestionsTopSearch = forwardRef<
...
       {...otherProps}
     >
-      <p className="suggestions__title">Top Search</p>
+      <div data-fs-search-suggestion-header>
+        <p data-fs-search-suggestion-title>Top Search</p>
+      </div>
       <UIList variant="ordered">
         {searchedItems.map((item, index) => (
-          <li key={index} className="suggestions__item">
+          <li key={index} data-fs-search-suggestion-item>
```
Comment (saranicoly) [inline src/components/search/Suggestions/SuggestionsTopSearch.tsx:44]: Same suggestion about not using the index as a key here.

**Review by saranicoly — COMMENTED** [review]
Great work! I left some comments.

Also, I think you forgot to modify the changelog?

File: src/sdk/search/useSearchHistory.ts [hunk 5]
```diff
@@ -5,9 +5,13 @@ const storageKey = 'main::store::searchHistory'
 const MAX_HISTORY_SIZE = 5
 
 export default function useSearchHistory(
+  history: string[] = [],
```
Comment (saranicoly) [inline src/sdk/search/useSearchHistory.ts:8]: Can you explain why did you add this?

File: src/components/search/History/SearchHistory.tsx (hunk 2 above)
Comment (filipewl) [reply src/components/search/History/SearchHistory.tsx]: Thanks for the article! I changed it [here](https://github.com/vtex-sites/gatsby.store/pull/7/commits/544924fcdf912f49199265a308e06ccc8fe59c36).

File: src/components/search/Suggestions/Suggestions.tsx (hunk 3 above)
Comment (filipewl) [reply src/components/search/Suggestions/Suggestions.tsx]: [Done](https://github.com/vtex-sites/gatsby.store/pull/7/commits/544924fcdf912f49199265a308e06ccc8fe59c36)!

File: src/sdk/search/useSearchHistory.ts (hunk 5 above)
Comment (filipewl) [reply src/sdk/search/useSearchHistory.ts:8]: I did this so we could show the `history` items in the `SearchHistory` component inside Storybook. 

This component uses `useStorage` to fetch this data, but inside Storybook there would be no way for us to populate this and the component being rendered with the items. 

This way, for the Storybook example, we can pass this initial data but it defaults to an empty array when not passed, which was already [the previous value](https://github.com/vtex-sites/gatsby.store/pull/7/files?file-filters%5B%5D=.ts&show-deleted-files=true&show-viewed-files=true#diff-912012eb2e9f77d2bb5c5c905a59393b2d844675034d4b61fc2e6959b529b885L10).
https://github.com/vtex-sites/gatsby.store/blob/e14452f08436a34dcb39e49412d5d8673943c09a/src/components/search/History/History.stories.tsx#L14

File: src/components/search/SuggestionProductCard/suggestion-product-card.scss [hunk 6]
```diff
@@ -3,6 +3,10 @@
 .suggestion-product-card {
```
Comment (hellofanny) [inline src/components/search/SuggestionProductCard/suggestion-product-card.scss:3]: I think we missed this class, should we update to `data-suggestion-product-card` ? 👀  (If so, remember to update  on [SuggestionProductCard.tsx](https://github.com/vtex-sites/gatsby.store/blob/b06bd2c58ed940bf7998adb91f208c267c3ba7f2/src/components/search/SuggestionProductCard/SuggestionProductCard.tsx#L33))

Comment (hellofanny) [inline src/components/search/SuggestionProductCard/suggestion-product-card.scss:3]:
```suggestion
[data-suggestion-product-card] {
```

File: src/components/search/SuggestionProductCard/suggestion-product-card.scss (hunk 1 above)
Comment (hellofanny) [reply src/components/search/SuggestionProductCard/suggestion-product-card.scss:7]: Nice!!! I think that is the easiest way to fix it, I'll see if there is another way to do it 💭

File: src/components/search/SuggestionProductCard/suggestion-product-card.scss (hunk 6 above)
Comment (filipewl) [reply src/components/search/SuggestionProductCard/suggestion-product-card.scss]: You're right! I've replaced the classes with `data-fs-*` [here](https://github.com/vtex-sites/gatsby.store/pull/7/commits/3802754f668a0b7f88cbcdd6e539177ccfefc92c).

Comment (filipewl) [reply src/components/search/SuggestionProductCard/suggestion-product-card.scss]: Addressed it [here](https://github.com/vtex-sites/gatsby.store/pull/7/commits/3802754f668a0b7f88cbcdd6e539177ccfefc92c).

**Comment by hellofanny** [comment]
Well done 💯 🥳 
I've noticed that svg issue on safari is still there. How about we replacing this component?

image

**Comment by filipewl** [comment]
> Well done 💯 🥳 I've noticed that svg issue on safari is still there. How about we replacing this component?
> 
> image

Thaaanks! Replaced it [here](https://github.com/vtex-sites/gatsby.store/pull/7/commits/85a73178a58601e7fb030b8ffd4a6c80714b4d59). It's now working on Safari.

CleanShot 2022-05-04 at 17 21 18@2x

**Review by eduardoformiga — APPROVED** [review]
LGTM, wands! 
Left some comments but not blocking.

File: src/components/search/History/SearchHistory.tsx [hunk 7]
```diff
@@ -20,36 +21,33 @@ const doSearch = (term: string) => {
...
-              <UIIcon
-                component={<Icon name="Clock" width={18} height={18} />}
+              <Icon
+                name="Clock"
+                width={18}
+                height={18}
+                data-fs-search-suggestion-item-icon
               />
               {item}
-              <UIIcon
```
[the remark this answers is not in the document]
Comment (filipewl) [reply src/components/search/History/SearchHistory.tsx:47]: In the newest version for the [design spec](https://www.figma.com/file/kJywgfIx8LBIzggCY5AeDy/Handoff-v3.1-(Search)?node-id=162%3A107919) the icon is gone so I went ahead and removed it.
😬 
CleanShot 2022-05-10 at 21 01 43

File: src/components/search/Suggestions/Suggestions.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+import type { SuggestionsProps } from '.'
+
+export default {
+  component: Suggestions,
+  title: 'Organisms/Search/Suggestions',
+}
+
+const Template = (props: SuggestionsProps) => (
+  <SessionProvider>
+    <Suggestions {...props} />
```
[the remark this answers is not in the document]
Comment (filipewl) [reply src/components/search/Suggestions/Suggestions.stories.tsx]: Good idea! [Done](https://github.com/vtex-sites/gatsby.store/pull/7/commits/a682e6af5f512d69762e35d50eec9ace0e002f6c).

CleanShot 2022-05-10 at 21 31 38

File: src/components/search/Suggestions/SuggestionsTopSearch.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,23 @@
+import { SuggestionsTopSearch } from '.'
+import type { SuggestionsTopSearchProps } from '.'
+
+export default {
+  component: SuggestionsTopSearch,
+  title: 'Organisms/Search/TopSearch',
+}
+
+const Template = (props: SuggestionsTopSearchProps) => (
+  <SuggestionsTopSearch {...props} />
```
[the remark this answers is not in the document]
Comment (filipewl) [reply src/components/search/Suggestions/SuggestionsTopSearch.stories.tsx]: Done. See [this previous comment](https://github.com/vtex-sites/gatsby.store/pull/7#discussion_r869776054).
